coralogix/resource_coralogix_group_v2.go

```diff
@@ -0,0 +1,899 @@
+		Filters: filters,
+	}
+
+	return types.ObjectValueFrom(ctx, ScopeAttrs(), scopeModel)
+}
+
+func flattenGroupScopeFilters(ctx context.Context, filters *cxsdk.ScopeFilters) (types.Object, diag.Diagnostics) {
+	if filters == nil {
+		return types.ObjectNull(ScopeFiltersAttrs()), nil
+	}
+
+	subsystems, diags := flattenGroupScopeFiltersList(ctx, filters.Subsystems)
+	if diags.HasError() {
+		return types.ObjectNull(ScopeFiltersAttrs()), diags
+	}
+
+	applications, diags := flattenGroupScopeFiltersList(ctx, filters.Applications)
+	if diags.HasError() {
+		return types.ObjectNull(ScopeFiltersAttrs()), diags
+	}
+
+	scopeFiltersModel := ScopeFiltersModel{
+		Subsystems:   subsystems,
+		Applications: applications,
+	}
+
+	return types.ObjectValueFrom(ctx, ScopeFiltersAttrs(), scopeFiltersModel)
+}
+
+func flattenGroupScopeFiltersList(ctx context.Context, filters []*cxsdk.GroupScopeFilter) (types.Set, diag.Diagnostics) {
+	if filters == nil {
+		return types.SetNull(types.ObjectType{AttrTypes: GroupScopeFilterAttrs()}), nil
+	}
+
+	flattenedFilters := make([]GroupScopeFilterModel, 0, len(filters))
+	for _, filter := range filters {
+		flattenedFilters = append(flattenedFilters, GroupScopeFilterModel{
+			Term:       types.StringValue(filter.Term),
+			FilterType: types.StringValue(FilterTypeProtoToSchema[filter.FilterType]),
+		})
+	}
+
+	return types.SetValueFrom(ctx, types.ObjectType{AttrTypes: GroupScopeFilterAttrs()}, flattenedFilters)
+}
+
+func flattenGroupId(id *cxsdk.TeamGroupID) types.String {
+	if id == nil {
+		return types.StringNull()
+	}
+
+	return types.StringValue(strconv.Itoa(int(id.Id)))
+}
+
+func flattenTeamId(id *cxsdk.GroupsTeamID) types.String {
+	if id == nil {
+		return types.StringNull()
+	}
+
+	return types.StringValue(strconv.Itoa(int(id.Id)))
+}
+
+func flattenScopeId(id *cxsdk.ScopeID) types.String {
+	if id == nil {
+		return types.StringNull()
+	}
+
+	return types.StringValue(strconv.Itoa(int(id.Id)))
+
+}
+
+type GroupV2ResourceModel struct {
+	ID             types.String `tfsdk:"id"`
+	Name           types.String `tfsdk:"name"`
+	TeamId         types.String `tfsdk:"team_id"`
+	Description    types.String `tfsdk:"description"`
+	ExternalId     types.String `tfsdk:"external_id"`
+	Roles          types.Set    `tfsdk:"roles"` // GroupRolesModel
+	Users          types.Set    `tfsdk:"users"` // UserModel
+	Scope          types.Object `tfsdk:"scope"` // ScopeModel
+	NextGenScopeId types.String `tfsdk:"next_gen_scope_id"`
+}
+
+type GroupRolesModel struct {
+	ID          types.String `tfsdk:"id"`
+	Name        types.String `tfsdk:"name"`
+	Description types.String `tfsdk:"description"`
+}
+
+func RolesAttrs() map[string]attr.Type {
+	return map[string]attr.Type{
+		"id":          types.StringType,
+		"name":        types.StringType,
+		"description": types.StringType,
+	}
+}
+
+type ScopeModel struct {
+	ID      types.String `tfsdk:"id"`
+	Filters types.Object `tfsdk:"filters"` // ScopeFiltersModel
```

Comment:
What's the reason you use an Object here? you could use the type directly as well, right?